worlds/undertale/__init__.py

```diff
@@ -99,110 +116,107 @@ def create_items(self):
+            if not self.options.no_equips:
                 itempool = ["Real Knife" if item == "Worn Dagger" else "The Locket"
                             if item == "Heart Locket" else item for item in itempool]
-        if self.multiworld.only_flakes[self.player]:
+        if self.options.only_flakes:
             itempool = [item for item in itempool if item not in non_key_items]
 
-        starting_key = self.multiworld.starting_area[self.player].current_key.title() + " Key"
+        starting_key = self.options.starting_area.current_key.title() + " Key"
         itempool.remove(starting_key)
         self.multiworld.push_precollected(self.create_item(starting_key))
         # Choose locations to automatically exclude based on settings
         exclusion_pool = set()
-        exclusion_pool.update(exclusion_table[self.multiworld.route_required[self.player].current_key])
-        if not self.multiworld.rando_love[self.player] or \
-                (self.multiworld.route_required[self.player] != "genocide" and
-                 self.multiworld.route_required[self.player] != "all_routes"):
+        exclusion_pool.update(exclusion_table[self.options.route_required.current_key])
+        if not self.options.rando_love or \
+                (self.options.route_required != "genocide" and
+                 self.options.route_required != "all_routes"):
             exclusion_pool.update(exclusion_table["NoLove"])
-        if not self.multiworld.rando_stats[self.player] or \
-                (self.multiworld.route_required[self.player] != "genocide" and
-                 self.multiworld.route_required[self.player] != "all_routes"):
+        if not self.options.rando_stats or \
+                (self.options.route_required != "genocide" and
+                 self.options.route_required != "all_routes"):
             exclusion_pool.update(exclusion_table["NoStats"])
 
         # Choose locations to automatically exclude based on settings
         exclusion_checks = set()
         exclusion_checks.update(["Nicecream Punch Card", "Hush Trade"])
         exclusion_rules(self.multiworld, self.player, exclusion_checks)
 
-        # Fill remaining items with randomly generated junk or Temmie Flakes
-        if not self.multiworld.only_flakes[self.player]:
-            itempool += self.multiworld.random.choices(list(junk_pool.keys()), weights=list(junk_pool.values()),
-                                                       k=len(self.location_names)-len(itempool)-len(exclusion_pool))
-        else:
-            itempool += ["Temmie Flakes"] * (len(self.location_names) - len(itempool) - len(exclusion_pool))
         # Convert itempool into real items
         itempool = [item for item in map(lambda name: self.create_item(name), itempool)]
+        # Fill remaining items with randomly generated junk or Temmie Flakes
+        while len(itempool) < len(self.multiworld.get_unfilled_locations(self.player)):
+            itempool += [self.create_filler()]
```

Comment:
**Extreme nitpick alert**

`itempool.append(self.create_filler())` is faster

*Low Priority*